lib/tests/test_logsource.c

```diff
@@ -415,4 +417,92 @@ Test(log_source, test_mangle_callback)
   test_source_destroy(source);
 }
 
+
+static DynamicWindowPool *
+test_dynamic_window_pool_init(gsize pool_size)
+{
+  DynamicWindowPool *pool = dynamic_window_pool_new(pool_size);
+  pool->balanced_window = pool_size;
+  dynamic_window_pool_init(pool);
+
+  return pool;
+}
+
+Test(log_source, test_dynamic_window_is_disabled_by_default)
+{
+  LogSource *source = test_source_init(&source_options);
+
+  cr_assert_not(log_source_is_dynamic_window_enabled(source));
+
+  test_source_destroy(source);
+}
+
+Test(log_source, test_dynamic_window)
+{
+  source_options.init_window_size = 0;
+
+  LogSource *source = test_source_init(&source_options);
+  log_source_set_name(source, "test-source-name");
+
+  cr_assert_not(log_source_free_to_send(source));
+
+  const gsize pool_size = 1000;
+  DynamicWindowPool *pool = test_dynamic_window_pool_init(pool_size);
+  log_source_enable_dynamic_window(source, pool);
+  cr_assert(log_source_is_dynamic_window_enabled(source));
+
+  /* currently unused */
+  log_source_dynamic_window_update_statistics(source);
+
+  pool->balanced_window = 500;
+  log_source_dynamic_window_realloc(source);
+  cr_assert(log_source_free_to_send(source),
+            "Source should not be suspended as it should own free dynamic window slots");
+  cr_assert_eq(pool->free_window, pool_size - pool->balanced_window);
```

Comment:
I think it would be better to use `pool->pool_size` instead of `pool_size`.

Reply:
OK.